kickstarter/android-oss
# WEB-665   [Android] When Facebook-only users log in to Kickstarter, require them to set a password

# 📲 What
When Facebook-only users log in to Kickstarter, require them to set a password


# 🤔 Why

Remove Facebook Login

# 🛠 How

Behind Feature Flag
As a User who does not have a Kickstarter password but can log in via Facebook
When I attempt to log in to Kickstarter using Facebook
I should be directed to set a new password
And if I try to navigate away without setting a password
I should be continually redirected to the set a password page until I have set one.

When I have successfully set a password, I should be directed as normal to the home page.

# 👀 See



https://user-images.githubusercontent.com/1075310/187357864-21ac9add-cdb3-4d68-93ef-25697d7bd39a.mp4

# 📋 QA

1. Add user Device id for feature flag "ANDROID_FACEBOOK_LOGIN_REMOVE" For production 
2. Remove the Facebook account Password with help from the "Web team" from database
3. Login with Facebook account and it would be nice if we test with 2FA and without 2FA
4. Check the app redirect to set Password Screen With user email masked
5. Kill app And reopen
6. Test Password different combination 
7. Set password and submit 
8. Check the app working normally  and can pledge projects 

 

# Story 📖

[[WEB-665]](https://kickstarter.atlassian.net/browse/WEB-665)


[WEB-665]: https://kickstarter.atlassian.net/browse/WEB-665?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
